cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -0,0 +1,168 @@
+    photon_os_password = credentials['vspherehost-nested'].get('password')
+    _configure_server_roles_for_test()  # make sure websocket is enabled
+    if provider.one_of(VMwareProvider):
+        _configure_vmware_console_for_test(appliance)
+
+    vm_obj.open_console(console='VM Console')
+    assert vm_obj.vm_console is not None, 'VMConsole object should be created'
+
+    vm_console = vm_obj.vm_console
+
+    # If the banner/connection-status element exists we can get
+    # the connection status text and if the console is healthy, it should connect.
+    assert vm_console.wait_for_connect() is True, "VM Console did not reach 'connected' state"
+
+    # Get the login screen image, and make sure it is a jpeg file:
+    screen = vm_console.get_screen()
+    assert imghdr.what('', screen) == 'jpeg'
+
+    # Wait is needed for Infra VMs which is not required for Openstack Instances
+    # Since infra VMs take more time to boot up
+    if not provider.one_of(OpenStackProvider):
+        logger.info("Wait 60 sec for 'login:' prompt")
+        time.sleep(60)  # wait for "login:" prompt
+
+    # Try to get the login screen text. We should have the text
+    # login there:
+    console_text = vm_console.get_screen_text()
+    assert re.search("login:", console_text)
+
+    # Enter Username:
+    # XXX: This is hard coded, we need to add some new yaml to track the
+    #      template credentials.
+    if provider.one_of(OpenStackProvider):
+        vm_console.send_keys("cirros\n")  # For Cirros Template
+    else:
+        vm_console.send_keys("{}\n".format(photon_os_username))  # For PhotonOS Template
+
+    # Check if console prompted for password
+    wait_for(func=lambda: re.search("Password:", vm_console.get_screen_text()),
+            delay=1,
+            num_sec=20)
+    console_text = vm_console.get_screen_text()
+    # Enter Password if prompted for password
+    assert re.search("Password:", console_text)
+
+    if provider.one_of(OpenStackProvider):
+        vm_console.send_keys("cubswin:)\n")  # For Cirros Template
+    else:
+        vm_console.send_keys("{}\n".format(photon_os_password))  # For PhotonOS Template
+
+    logger.info("Wait 15 secs to get the '$' prompt")
+    time.sleep(15)  # Wait for "$" prompt
+    # create file on system
+    vm_console.send_keys("touch blather\n")
+    logger.info("Wait 5 sec for file creation")
+    time.sleep(5)  # Wait for file creation
+
+    # Test pressing ctrl-alt-delete...we should be able to get a new login prompt:
+    vm_console.send_ctrl_alt_delete()
+    logger.info("Wait 60 sec to get 'login:' prompt since ctrl+alt+del resets VM/Instance")
+    time.sleep(60)   # Wait for "login" prompt
+    console_text = vm_console.get_screen_text()
+    assert re.search("login:", console_text)
+    # Code below only works for VMware and not tested against other providers
+    connect_ip, tc = wait_for(provider.mgmt.get_ip_address, [vm_obj.name], num_sec=300,
```

Comment:
It only works for VMware because for OpenStack, I get ip address as "None", so I am trying to figure out why. So far, I have understood, correct me if I am wrong, but I will need a floating IP to be associated to OpenStack instance then it will not return None for get_ip_address. Also, this code might work as is with the RHEV, not tested, hence not put conditional.